osu.Game/Screens/Select/Carousel/CarouselItem.cs

```diff
@@ -43,15 +41,15 @@ protected CarouselItem()
         /// <summary>
         /// Create a fresh drawable version of this item.
         /// </summary>
-        public abstract DrawableCarouselItem CreateDrawableRepresentation();
+        public abstract DrawableCarouselItem? CreateDrawableRepresentation();
 
         public virtual void Filter(FilterCriteria criteria)
         {
         }
 
         public virtual int CompareTo(FilterCriteria criteria, CarouselItem other) => ItemID.CompareTo(other.ItemID);
 
-        public int CompareTo(CarouselItem other) => CarouselYPosition.CompareTo(other.CarouselYPosition);
+        public int CompareTo(CarouselItem? other) => CarouselYPosition.CompareTo(other?.CarouselYPosition);
```

Comment:
This incurs boxing overhead as the `CompareTo(object?)` overload is used rather than `CompareTo(float)` as previously, which seems not great in a comparator function. I'd go the long-winded route here.

Not that I'm sure this comparison is actually ever used anywhere, mind...

Reply:
i figured this would be fine since the same thing has been done in other files, not sure how exactly the long winded route would be done, but ill look into it https://github.com/ppy/osu/blob/master/osu.Game/Online/Spectator/SpectatorScoreProcessor.cs#L187

Reply:
I'll fix it myself and fast-track this.